tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/i3d_evaluator.py

```diff
@@ -0,0 +1,411 @@
+            if self.input_blob is None:
+                output_blob = next(iter(self.exec_network.outputs))
+            else:
+                output_blob = (
+                    '_'.join([self.net_type, self.output_blob])
+                    if with_prefix else self.output_blob.split('{}_'.format(self.net_type))[-1]
+                )
+            self.input_blob = input_blob
+            self.output_blob = output_blob
+            self.with_prefix = with_prefix
+
+    def print_input_output_info(self):
+        print_info('{} - Input info:'.format(self.net_type))
+        has_info = hasattr(self.network if self.network is not None else self.exec_network, 'input_info')
+        if self.network:
+            if has_info:
+                network_inputs = OrderedDict(
+                    [(name, data.input_data) for name, data in self.network.input_info.items()]
+                )
+            else:
+                network_inputs = self.network.inputs
+            network_outputs = self.network.outputs
+        else:
+            if has_info:
+                network_inputs = OrderedDict([
+                    (name, data.input_data) for name, data in self.exec_network.input_info.items()
+                ])
+            else:
+                network_inputs = self.exec_network.inputs
+            network_outputs = self.exec_network.outputs
+        for name, input_info in network_inputs.items():
+            print_info('\tLayer name: {}'.format(name))
+            print_info('\tprecision: {}'.format(input_info.precision))
+            print_info('\tshape {}\n'.format(input_info.shape))
+        print_info('{} - Output info'.format(self.net_type))
+        for name, output_info in network_outputs.items():
+            print_info('\tLayer name: {}'.format(name))
+            print_info('\tprecision: {}'.format(output_info.precision))
+            print_info('\tshape: {}\n'.format(output_info.shape))
+
+    def fit_to_input(self, input_data):
+        has_info = hasattr(self.exec_network, 'input_info')
+        input_info = (
+            self.exec_network.input_info[self.input_blob].input_data
+            if has_info else self.exec_network.inputs[self.input_blob]
+        )
+        input_data = np.array(input_data)
+        input_data = np.transpose(input_data, (3, 0, 1, 2))
+        input_data = np.reshape(input_data, input_info.shape)
+        return {self.input_blob: input_data}
+
+    def prepare_data(self, data):
+        pass
+
+
+class I3DRGBModel(BaseModel):
+    def __init__(self, network_info, launcher, data_source, delayed_model_loading=False):
+        self.net_type = 'rgb'
+        super().__init__(network_info, launcher, data_source, delayed_model_loading)
+
+    def prepare_data(self, data):
+        video = Path(data[0].video)
+        frames_identifiers = [video / frame for frame in data[0].frames]
+        prepared_data = self.reader(frames_identifiers)
```

Comment:
why does this part can not be inside the reader. If I right understand, you just repack clip identifier logic?